memgpt/agent.py

```diff
@@ -518,7 +518,8 @@ def handle_ai_response(self, response_message):
             self.interface.function_message(f"Running {function_name}({function_args})")
             try:
                 function_args["self"] = self  # need to attach self to arg since it's dynamically linked
-                function_response_string = function_to_call(**function_args)
+                function_response = function_to_call(**function_args)
+                function_response_string = validate_function_response(function_response)

```

Comment:
We are now validating function outputs before putting them into a function response message (in `agent.messages`)